sdk/core/Azure.Core/tests/CloudEventTests.cs

```diff
@@ -0,0 +1,673 @@
+        {
+            var time = DateTimeOffset.Now;
+            var data = new TestModel
+            {
+                A = 10,
+                B = true
+            };
+            var dataSerializer = new JsonObjectSerializer(new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase
+            });
+            var cloudEvent = new CloudEvent("source", "type", dataSerializer.Serialize(data), "application/json", CloudEventDataFormat.Json)
+            {
+                Subject = "subject",
+                DataSchema = "schema",
+                Id = "id",
+                Time = time,
+            };
+            var serializer = new JsonObjectSerializer();
+            BinaryData serialized = serializer.Serialize(cloudEvent);
+            CloudEvent deserialized = CloudEvent.Parse(serialized.ToString())[0];
+            Assert.AreEqual("source", deserialized.Source);
+            Assert.AreEqual("type", deserialized.Type);
+            Assert.AreEqual(10, deserialized.Data.ToObject<TestModel>(dataSerializer).A);
+            Assert.AreEqual(true, deserialized.Data.ToObject<TestModel>(dataSerializer).B);
+            Assert.AreEqual("subject", deserialized.Subject);
+            Assert.AreEqual("schema", deserialized.DataSchema);
+            Assert.AreEqual("id", deserialized.Id);
+            Assert.AreEqual(time, deserialized.Time);
+
+            deserialized = (CloudEvent)serializer.Deserialize(serialized.ToStream(), typeof(CloudEvent), CancellationToken.None);
+            Assert.AreEqual("source", deserialized.Source);
+            Assert.AreEqual("type", deserialized.Type);
+            Assert.AreEqual(10, deserialized.Data.ToObject<TestModel>(dataSerializer).A);
+            Assert.AreEqual(true, deserialized.Data.ToObject<TestModel>(dataSerializer).B);
+            Assert.AreEqual("subject", deserialized.Subject);
+            Assert.AreEqual("schema", deserialized.DataSchema);
+            Assert.AreEqual("id", deserialized.Id);
+            Assert.AreEqual(time, deserialized.Time);
+
+            deserialized = CloudEvent.Parse(serialized);
+            Assert.AreEqual("source", deserialized.Source);
+            Assert.AreEqual("type", deserialized.Type);
+            Assert.AreEqual(10, deserialized.Data.ToObject<TestModel>(dataSerializer).A);
+            Assert.AreEqual(true, deserialized.Data.ToObject<TestModel>(dataSerializer).B);
+            Assert.AreEqual("subject", deserialized.Subject);
+            Assert.AreEqual("schema", deserialized.DataSchema);
+            Assert.AreEqual("id", deserialized.Id);
+            Assert.AreEqual(time, deserialized.Time);
+        }
+
+        [Test]
+        [TestCase("hello world")]
+        [TestCase("\"hello world\"")]
+        [TestCase("{\"key\":1}")]
+        [TestCase("[{key:1}]")]
+        [TestCase("<much wow=\"xml\"/>")]
+        public void CanRoundTripString(string data)
```

Comment:
We need a method like this for JSON. With all kinds of supported JSON values. 1/"s"/null etc.